json/DataTypes/DecodeStudentLong.roc

```diff
@@ -0,0 +1,81 @@
+module [getListStudents]
+
+import JsonData exposing [Json]
+import Student exposing [Student, Module]
+
+JsonErrors : [
+    FieldNotFound Str,
+    ExpectedJsonObject Str,
+    ExpectedJsonArray Str,
+]
+
+getListStudents : Json -> Result (List Student) JsonErrors
+getListStudents = \json ->
+    when json is
+        Object obj ->
+            when Dict.get obj "students" is
+                Ok students -> checkStudentList students
+                _ -> Err (FieldNotFound "Expected field with name students in object")
+
+        _ -> Err (ExpectedJsonObject "Expected an Object")
+
+checkStudentList : Json -> Result (List Student) JsonErrors
+checkStudentList = \json ->
+    when json is
+        Arr students ->
+            List.mapTry (List.map students checkStudentObj) \x -> x
+
+        _ -> Err (ExpectedJsonArray "Expected an Array")
+
+checkStudentObj : Json -> Result Student JsonErrors
+checkStudentObj = \json ->
+    when json is
+        Object obj ->
+            when Dict.get obj "name" is
+                Ok (String name) ->
+                    when Dict.get obj "modules" is
+                        Ok modules ->
+                            when checkModulesList modules is
+                                Ok goodMods ->
+                                    when Dict.get obj "finalGrade" is
+                                        Ok (Number fg) -> Ok (GraduatedStudent { name: name, modules: goodMods, finalGrade: fg })
+                                        _ ->
+                                            when Dict.get obj "currentGrade" is
+                                                Ok (Number cg) -> Ok (CurrentStudent { name: name, modules: goodMods, currentGrade: cg })
+                                                _ -> Err (FieldNotFound "Expected field ")
```

Comment:
This is awesome, great work coming up with a strong motivating example for composable decoders! 😂